include/logging/log_core.h

```diff
@@ -238,9 +238,9 @@ extern "C" {
 		if (_LOG_CONST_LEVEL_CHECK(_level) &&			    \
 		    (_level <= LOG_RUNTIME_FILTER(_filter))) {		    \
 			struct log_msg_ids src_level = {		    \
-				.level = _level,			    \
-				.source_id = _id,			    \
-				.domain_id = CONFIG_LOG_DOMAIN_ID	    \
```

Comment:
try just changing the ordering here so that you initialize the members in the same order they are in the struct definition, it should work in C++